### DISCUSSION TITLE
How to submit a form with post?

### DISCUSSION THREAD

**Discussion by davyzhang** [sagebind/isahc#168] [Questions] [answered]
It's quite a common requirement, but I couldn't find any documentation of it.

**Comment by sagebind** [comment, marked as answer]
Submitting a form is just sending a `POST` request with a particularly-formatted body. Isahc currently does not have an API to generate this body from form parameters for you at the moment, see #14. For now, there might be other crates you could use to generate the form body, convert it to a byte array, and then send that.

Sending a `POST` request can be done with functions like [`post`](https://docs.rs/isahc/0.8.2/isahc/fn.post.html) or [`HttpClient::post`](https://docs.rs/isahc/0.8.2/isahc/struct.HttpClient.html#method.post).

**Comment by ta3pks** [comment]
@davyzhang  What i do in general when i need a custom request is to create a new http request using [http](https://crates.io/crates/http) and send it using isahc

**Comment by davyzhang** [comment]
Thanks, @NikosEfthias @sagebind. I understand the idea now.
